### PR TITLE
Utility method for length estimation of utf8

### PR DESCRIPTION
**Pull request by navis**
Simple estimation on length of utf8 string

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

**Comment by himanshug** [comment]
can you pls have a short description in the PR about why it is needed?

**Comment by drcrallen** [comment]
Before merging, it would be nice to be able to document this change more.

If people are relying on the prior behavior to estimate data "size", then this change could make subsequent computations of that size to be different from prior versions.

Support for the prior methodology should be retained (and potentially deprecated) unless tests are in place to ensure they return the same results.

(optional) The static method would probably fit under java-util StringUtils better than GuavaUtils

**Comment by navis** [comment]
@himanshug to remove overhead of encoding? UTF8 makes byte[] which is 3x of string.length() and copies again to make final result. If it's just for estimation, we can skip that part.

@drcrallen Yes, some UTs would be good and I also agree on `StringUtils` is right place to be. Really really wish someone move api and utils included in druid itself.

**Comment by navis** [comment]
Moved to java-util (https://github.com/metamx/java-util/pull/45)

**Comment by himanshug** [comment]
this is possibly going to break the integration tests which make segmentMetadata queries, pls check.

**Comment by navis** [comment]
@himanshug confirmed result is not changed. testcase is included.

File: common/src/main/java/io/druid/common/guava/GuavaUtils.java [hunk 1]
```diff
@@ -95,4 +95,24 @@ public InputStreamReader getInput() throws IOException
       }
     };
   }
+
+  // should be used only for estimation.. does not check validity of format
+  public static int binaryLengthAsUTF8(String value)
```
Comment (xvrl) [inline common/src/main/java/io/druid/common/guava/GuavaUtils.java]: any reason we put this in GuavaUtils? It does not really seem to be related to Guava.

**Comment by xvrl** [comment]
@navis I think it makes sense adding a string utility class to druid to make it easier to do those kinds of changes. A lot of String optimizations in that area are going to be druid specific anyway.

File: common/src/main/java/io/druid/common/guava/GuavaUtils.java (hunk 1 above)
Comment (navis) [inline common/src/main/java/io/druid/common/guava/GuavaUtils.java]: @xvrl moved to `StringUtils` in java-util (metamx/java-util#45)

**Comment by navis** [comment]
@xvrl Could I understand that as this would be better in druid, not in java-util?

**Comment by xvrl** [comment]
@navis yes, I think it doesn't need to live in java-util since this is only used by Druid itself.

**Comment by navis** [comment]
@xvrl rebased and added test cases

File: common/src/main/java/io/druid/common/utils/StringUtils.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+package io.druid.common.utils;
+
+/**
+ */
+public class StringUtils extends com.metamx.common.StringUtils
+{
+  // should be used only for estimation
+  // returns the same result with StringUtils.fromUtf8(value).length for valid string values
+  // does not check validity of format and returns over-estimated result for invalid string (see UT)
+  public static int binaryLengthAsUTF8(String value)
```
Comment (gianm) [inline common/src/main/java/io/druid/common/utils/StringUtils.java]: Please rename to `estimatedBinaryLengthAsUTF8` if this should only be used for estimation; will make things much clearer for future developers.

**Comment by gianm** [comment]
:+1: after the renaming comment.

@navis out of curiosity do you use this "size" analysis type for anything? I had thought it was not really that useful and I assumed most people issuing segment metadata queries disable it.

**Comment by navis** [comment]
@gianm Yes, I don't think we can use it meaningfully but it can still make someone happy (UI guys want to show numbers whenever it's possible). By the way I'm thinking of using this to estimate the size of incremental index to be flushed automatically rather than row count. Ah, I've forgot I've already done that in #2459. You can check that if interested.

File: common/src/main/java/io/druid/common/utils/StringUtils.java (hunk 2 above)
Comment (navis) [inline common/src/main/java/io/druid/common/utils/StringUtils.java]: missed that. consider it done.

**Pull request merged**